flycheck.el

```diff
@@ -5264,7 +5264,16 @@ function `buffer-file-name'."
       (when (seq-some (apply-partially #'flycheck-same-files-p
                                        (expand-file-name filename cwd))
                       buffer-files)
-        (setf (flycheck-error-filename err) (buffer-file-name)))))
+        (let ((new-filename (buffer-file-name)))
+          (setf (flycheck-error-filename err) new-filename
+                (flycheck-error-message err)
+                (replace-regexp-in-string
+                 (regexp-quote filename)
+                 new-filename
+                 (flycheck-error-message err)
+                 nil
+                 t ;; do literal substitution
```

Comment:
Instead of `t` + a comment you can use a non-nil symbol, e.g. `'literally`, I think.